internal/identity/manager/manager.go

```diff
@@ -288,9 +268,12 @@ func (mgr *Manager) mergeGroups(ctx context.Context, directoryGroups []*director
 				}
 				defer mgr.dataBrokerSemaphore.Release(1)
 
-				_, err = mgr.cfg.Load().dataBrokerClient.Delete(ctx, &databroker.DeleteRequest{
-					Type: any.GetTypeUrl(),
-					Id:   id,
+				_, err = mgr.cfg.Load().dataBrokerClient.Put(ctx, &databroker.PutRequest{
+					Record: &databroker.Record{
+						Type:      any.GetTypeUrl(),
+						Id:        id,
+						DeletedAt: timestamppb.Now(),
+					},
 				})
 				if err != nil {
 					return fmt.Errorf("failed to delete directory group: %s", id)
```

Comment:
this error doesn't seem to be visible anywhere, no? there's no `eg.Wait()`

Reply:
There was supposed to be an `eg.Wait`. I added it.